lib/Backend/GlobOpt.cpp

```diff
@@ -10835,7 +10820,10 @@ GlobOpt::TypeSpecializeIntUnary(
     IR::Opnd *src1 = instr->GetSrc1();
     this->ToInt32(instr, src1, this->currentBlock, src1ValueToSpecialize, nullptr, lossy);
 
-    if(bailOutKind != IR::BailOutInvalid && !this->IsLoopPrePass())
+    bool ignoreOverflowBailout = CONFIG_FLAG(RemoveIntOverflow) && instr->GetSrc1()->GetType() == TyInt32 && bailOutKind == IR::BailOutOnOverflow;
```

Comment:
this doesn't seem like the right place... seems better for instance to have Instr::ShouldCheckForIntOverflow() check for this flag